In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import geopandas as gpd
import folium

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [49]:
## 공원 정제 전 데이터 (모든 공원 정보 다 들어있음)
parkdata = '/content/drive/MyDrive/jeonjusi/parkdata.csv'
parkdata = pd.read_csv(parkdata)

In [17]:
## 공원 정제 후 데이터 (운동시설 만)
park = '/content/drive/MyDrive/jeonjusi/park.csv'
park = pd.read_csv(park)

In [20]:
park.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 135 entries, 0 to 134
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   dataTitle     135 non-null    object 
 1   parkArea      135 non-null    float64
 2   parkFact4     135 non-null    object 
 3   parkGubun     135 non-null    object 
 4   parkLoadAddr  135 non-null    object 
 5   posx          135 non-null    float64
 6   posy          135 non-null    float64
 7   parkFact4-1   135 non-null    object 
 8   parkFact4-2   71 non-null     object 
 9   parkFact4-3   20 non-null     object 
 10  parkFact4-4   10 non-null     object 
 11  parkFact4-5   2 non-null      object 
dtypes: float64(3), object(9)
memory usage: 12.8+ KB


In [19]:
print(park.describe(), park.head())

           parkArea        posx        posy
count  1.350000e+02  135.000000  135.000000
mean   5.991719e+04  127.125075   35.823252
std    3.308586e+05    0.026998    0.019638
min    3.332000e+02  127.055235   35.783339
25%    1.552000e+03  127.105076   35.808182
50%    2.280000e+03  127.121180   35.825420
75%    1.003300e+04  127.141603   35.835403
max    3.373838e+06  127.177274   35.875452   dataTitle  parkArea          parkFact4 parkGubun  \
0      세병공원  200481.0  족구장,풋살장,농구장,다목적구장      근린공원   
1     원장동공원   16103.7               운동기구      근린공원   
2      하늘공원   25864.0           농구대,운동기구      근린공원   
3   아중18호공원    1500.0               운동기구     어린이공원   
4     나들목공원   56941.0               운동기구      근린공원   

                  parkLoadAddr        posx       posy parkFact4-1 parkFact4-2  \
0   전북특별자치도 전주시 덕진구 송천동2가 1316  127.120200  35.871980         족구장         풋살장   
1      전북특별자치도 전주시 덕진구 장동 1065  127.055235  35.841451        운동기구         NaN   
2    전북특별자치도 전주시 덕진구 덕진동2가 682  127.

## 지도 불러오기
- 운동시설이 있는 공원에만 마킹

In [24]:
# 유효한 위도(y)와 경도(x)를 사용합니다.
latitude = 35.8242238
longitude = 127.1479532
jeonju_map = folium.Map(location=[latitude, longitude], zoom_start=13)
jeonju_map


In [25]:
from folium.features import CustomIcon

In [55]:
# 데이터프레임을 반복하여 마커 추가
for _, row in park.iterrows():
    icon = folium.Icon(icon='glyphicon-star', color='green')  # Folium 내장 아이콘 사용
    popup = folium.Popup(f"Title: {row['dataTitle']}", max_width=200)
    folium.Marker(location=[row['posy'], row['posx']], icon=icon, popup=popup).add_to(jeonju_map)


In [56]:
jeonju_map

- 공원 전부 표시

In [51]:
# 'posx'와 'posy' 열에서 빈 문자열을 NaN으로 변경
parkdata['posx'] = pd.to_numeric(parkdata['posx'], errors='coerce')
parkdata['posy'] = pd.to_numeric(parkdata['posy'], errors='coerce')

# NaN 값이 있는 행을 제거 (필요한 경우)
parkdata= parkdata.dropna(subset=['posx', 'posy'])

In [52]:
parkdata.info()

<class 'pandas.core.frame.DataFrame'>
Index: 242 entries, 0 to 248
Data columns (total 23 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     242 non-null    int64  
 1   dataContent    242 non-null    object 
 2   dataSid        242 non-null    int64  
 3   dataTitle      242 non-null    object 
 4   parkAddr       242 non-null    object 
 5   parkArea       242 non-null    float64
 6   parkFact1      242 non-null    object 
 7   parkFact2      242 non-null    object 
 8   parkFact3      242 non-null    object 
 9   parkFact4      242 non-null    object 
 10  parkFact5      242 non-null    object 
 11  parkGubun      242 non-null    object 
 12  parkLoadAddr   242 non-null    object 
 13  parkLocation   242 non-null    object 
 14  parkNo         242 non-null    object 
 15  parkNoticeday  242 non-null    object 
 16  parkOrgnm      242 non-null    object 
 17  parkRegdate    242 non-null    object 
 18  parkSido       

In [53]:
# 데이터프레임을 반복하여 마커 추가
for _, row in parkdata.iterrows():
    icon = folium.Icon(icon='glyphicon-star', color='red')  # Folium 내장 아이콘 사용
    popup = folium.Popup(f"Title: {row['dataTitle']}", max_width=200)
    folium.Marker(location=[row['posy'], row['posx']], icon=icon, popup=popup).add_to(jeonju_map)


In [54]:
jeonju_map

- parkdata 가지고 먼저 지도에 띄우고, park 띄우면 됨 242개
- 초록색이 운동시설 있는 공원 135개